src/IonEvent.ts

```diff
@@ -404,12 +404,17 @@ class IonStructEvent extends AbsIonContainerEvent {//no embed support as of yet.
     structsEqual(actualEvents : AbstractIonEvent[], expectedEvents : AbstractIonEvent[]) : boolean {
         let matchFound : boolean = true;
         let paired : boolean[] = new Array<boolean>(expectedEvents.length);
-        for(let i : number = 0; matchFound && i < actualEvents.length; i++) {
+        for (let i : number = 0; matchFound && i < actualEvents.length; i++) {
             matchFound = false;
-            for(let j : number = 0; !matchFound && j < expectedEvents.length; j++) {
-                if(!paired[j]) {
+            for (let j : number = 0; !matchFound && j < expectedEvents.length; j++) {
+                if (!paired[j]) {
                     matchFound = actualEvents[i].equals(expectedEvents[j]);
-                    paired[j] = matchFound;
+                    if (matchFound && actualEvents[i].eventType === IonEventType.CONTAINER_START) {
+                        for (let k = j; k < expectedEvents[j].ionValue.length; k++) {
+                            paired[k] = true;
+                        }
+                        i += actualEvents[i].ionValue.length;
```

Comment:
will this cause some events to go unchecked?  i haven't been able to show that this is a problem, but my concern is similar to that raised below for lists/sexps

Reply:
the issue was in lists and sexps where the .length call was returning undefined and the += was translating i to NaN. It is now fixed.